### PR TITLE
#112: property groups

### LINKED ISSUE #112: property otc groups
the first level would show types as folders

the context menu on each type would have a "group by" sub menu which lists out all of the fields. When you select a field, each value in that field becomes a folder.

You can keep slicing it down more and more nested.

each folder should have a context menu option to "clear groupings" which collapses the nesting below it, removing the sub-folders

----

future work could have something like "re-assign" to a folder which would allow for mass-editing of metadata.

For example, right now, I have `type = health` and under that I have `what = note | action | treatment | meditation`. Meditation doesn't really fit into `health` anymore. I would like to group `health` by `what` and then right click `meditation` and change all of those to `type = meditation`.

### PR DESCRIPTION
**Pull request by msphair**
fixes #112 

- adds group icons and adjusts padding
- hides collapse icons when not hovering
- create new `OtcContainer` called `PropertyContainer` which shows property values for a selected property field
- differentiate sort options for `PropertyContainer`s and `TagGroupContainer`s
- Adds a new modal similar to the tag selector modal that selects a property field
- removes specific rendering of rx containers for a generic call using group type
- Adds menu to add tag group button to add either tag group or property group
- Change variable names to match the generic otc group vs specifically tag group

### CODE REVIEW

File: src/oblogger_view.ts [hunk 1]
```diff
@@ -794,12 +847,17 @@ export class ObloggerView extends ItemView {
         // Dump whatever remains (hopefully not much)
         this.otcGroupsDiv?.empty();
 
+        // todo: need to add a separator somewhere in here?
```
Comment (msphair) [inline src/oblogger_view.ts:861]: @robotbrad do we want to figure out UI for separating property groups from tag groups in this PR or in a follow up PR?

**Comment by robotbrad** [comment]
there seems to be an issue when adding a property group that is associated with a "property list" or "property array", not sure how best to describe it. adding one of these types of broken properties will break the rendering of all property groups.

it's fairly easy to reproduce
1. create a new note
2. add a property called "stuff"
3. change the property type to list
4. add "one" to the value of the "stuff" property, press enter
5. add "stuff" to oblogger
6. observe that there are no errors in the console or oblogger
7. add a second value called "two" to the "stuff" property
8. observe that oblogger rendering should break and note the error in the console log

Screenshot from 2023-09-25 08-45-16

**Comment by robotbrad** [comment]
adding a property group that contains URLs has unexpected folder rendering

Screenshot from 2023-09-25 09-00-26

edit: upgraded to issue #133

**Comment by robotbrad** [comment]
what do you think of adding a sort-otc-group-type setting to oblogger settings that allows otc groups to be mixed or separated by group types

**Comment by msphair** [comment]
> there seems to be an issue when adding a property group that is associated with a "property list" or "property array", not sure how best to describe it. adding one of these types of broken properties will break the rendering of all property groups.

this should be resolved now

**Comment by msphair** [comment]
> adding a property group that contains URLs has unexpected folder rendering

not going to fix this in this PR

**Comment by msphair** [comment]
> what do you think of adding a sort-otc-group-type setting to oblogger settings that allows otc groups to be mixed or separated by group types

I like the idea. Let's make a ticket for it and add it as a feature for after this pr

File: src/oblogger_view.ts (hunk 1 above)
Comment (robotbrad) [reply src/oblogger_view.ts:861]: expanded in #134

File: src/containers/otc/property_container.ts [hunk 2]
```diff
@@ -0,0 +1,143 @@
+import { ContainerCallbacks } from "../container_callbacks";
+import { ContainerSortMethod, ObloggerSettings, OtcGroupType } from "../../settings";
+import { App, Menu, TFile } from "obsidian";
+import { OtcContainer } from "./otc_container";
+import { FileState } from "../../constants";
```
Comment (robotbrad) [inline src/containers/otc/property_container.ts:5]: bookmarked files aren't sorting as expected. should be sorted to the top of their respective group/folder.

all other aspect of bookmarking in property groups is working as expected.

Comment (msphair) [reply src/containers/otc/property_container.ts:5]: bookmarks should be good now

**Review by robotbrad — APPROVED** [review]

**Pull request merged**